catalog/db_diff_scanner.go

```diff
@@ -82,44 +85,48 @@ func (s *DiffScanner) diffFromParent(tx db.Tx, params doDiffParams) (*DiffScanne
 	if err != nil {
 		return nil, fmt.Errorf("get child last commit failed: %w", err)
 	}
-	scannerOpts := DBScannerOptions{
-		After:            params.After,
-		AdditionalFields: prepareDiffAdditionalFields(params.AdditionalFields),
+	leftLineage, err := getLineage(tx, params.LeftBranchID, CommittedID)
+	if err != nil {
+		return nil, err
 	}
-	s.leftScanner = NewDBLineageScanner(tx, params.LeftBranchID, CommittedID, scannerOpts)
-	s.rightScanner = NewDBLineageScanner(tx, params.RightBranchID, UncommittedID, scannerOpts)
-	s.childLineage, err = getLineage(tx, params.RightBranchID, UncommittedID)
+	rightLineage, err := getLineage(tx, params.RightBranchID, UncommittedID)
 	if err != nil {
 		return nil, err
 	}
+	// If some ancestor branch commit id is the same for parent and child - then the parent does not need to read it
+	// so it is trimmed from the parent lineage
+
+	for i := range leftLineage {
```

Comment:
you removed the len check - don't want to verify that the child right side is bigger?

Reply:
done